assertj-core/src/test/java/org/assertj/core/api/hashset/HashSetAssert_contains_Test.java

```diff
@@ -0,0 +1,44 @@
+package org.assertj.core.api.hashset;
+
+import org.assertj.core.api.HashSetAssertBaseTest;
+import org.junit.jupiter.api.Test;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.time.Instant.EPOCH;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldContain.shouldContain;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.Sets.newHashSet;
+
+class HashSetAssert_contains_Test extends HashSetAssertBaseTest {
+  @Test
+  void should_pass() {
+    // WHEN/THEN
+    assertThatNoException()
+      .isThrownBy(() -> assertThat(newHashSet(Set.of("Yoda", "Luke", "Han"))).contains("Han", "Luke"));
+  }
+
+  @Test
+  void should_fail_if_missing_but_also_if_hashCode_changed() {
```

Comment:
Having separate tests for missing element and different hash code would be better

Reply:
Right, I'll ultimately split this, but wanted to emphasize the current behavior - first fail doesn't discover the changed `hashCode` and occurs just because of `missing` element. 

Not sure if this is the way to go - one might get a failure, fix something in code/tests and then get another failure (due to `hashCode` change). Ideally, fail should discover everything I think, but for that I might need to further improve the `ComparisonStrategy` perhaps to include both checks at once or do tricks like
```java
assertAll(() -> super.containsForProxy(values), () -> hashSets.assertContains(info, actual, values));
```
in the overriden `containsForProxy` 🤔

Reply:
why not use `doesNotContain` to emphasize the missing value is not in the hash set ?